RHDEVDOCS-3752 Document Pipelines as code (#43308)
openshift/openshift-docs: modules/op-customizing-pipelines-as-code-configuration.adoc

> // This module is included in the following assembly:

Author: I created them with the thought that I can use them in subsequent iterations of this document, when they are gonna be supported. I hoped that they will not mess with the build process and will not show up anywhere, as they have been commented out in the assembly. But now I deleted them - creating them again is cheap!

//
// *cicd/pipelines/using-pipelines-as-code.adoc

:_content-type: REFERENCE
[id="customizing-pipelines-as-code-configuration_{context}"]
= Customizing {pac} configuration 

[role="_abstract"]
To customize {pac}, cluster administrators can configure the following parameters using the `pipelines-as-code` config map in the `pipelines-as-code` namespace:

.Customizing {pac} configuration
[options="header"]
|===

| Parameter | Description | Default

> | `application-name` | The name of the application. For example, the name displayed in the GitHub Checks labels. | `"Pipelines as Code CI"` 

Reviewer: Do you mean for `"Pipelines as Code CI"` to be quoted this way?

Author: Yes, because the default value must be written as a string, including quotes.

> | `max-keep-days` | The number of the days for which the executed pipeline runs are kept in the `pipelines-as-code namespace`. 

Reviewer 2: The default value is missing from this line in the table.

Author: there is no default value

> Note that this configmap setting does not affect the cleanups of a user’s pipeline runs, which are controlled by the annotations on the pipeline run definition in the user’s GitHub repository. |

Reviewer: ```suggestion
Note that this configmap setting does not affect the cleanups of a user's pipeline runs, which are controlled by the annotations on the pipeline run definition in the user's GitHub repository. |
```
Avoid curly quotes and apostrophe's.

> Note that this configmap setting does not affect the cleanups of an user’s pipeline runs, which are controlled by the annotations on the pipeline run definition in the user’s GitHub repository. |

Reviewer 2: ```suggestion
Note that this configmap setting does not affect the cleanups of a user’s pipeline runs, which are controlled by the annotations on the pipeline run definition in the user’s GitHub repository. |
```
- is `cleanups` the correct technical term here? Is it referring to the name of a specific feature or function? If not, then the sentence should be re-worded to say something like `does not affect how a user's pipeline runs are cleaned up`

| `secret-auto-create` | Whether to auto create a secret with the token generated via the Github application to be used with private repositories. | `enabled` 

> | `remote-tasks` | Allows remote tasks from pipeline run annotations. | `enabled`

Reviewer 2: ```suggestion
| `remote-tasks` | When enabled, allows remote tasks from pipeline run annotations. | `enabled`
```

> | `hub-url` | The base url for the link:https://api.hub.tekton.dev/v1[Tekton Hub API]. | `https://hub.tekton.dev/`

Reviewer: ```suggestion
| `hub-url` | The base URL for the link:https://api.hub.tekton.dev/v1[Tekton Hub API]. | `https://hub.tekton.dev/`
```
